javalib/src/main/scala/java/net/PlainSocketImpl.scala

```diff
@@ -0,0 +1,463 @@
+package java.net
+
+import scala.scalanative.native._
+import scala.scalanative.runtime.ByteArray
+import scala.scalanative.posix.errno._
+import scala.scalanative.posix.sys.socket
+import scala.scalanative.posix.sys.socketOps._
+import scala.scalanative.posix.netinet.in
+import scala.scalanative.posix.netinet.inOps._
+import scala.scalanative.posix.netinet.tcp
+import scala.scalanative.posix.arpa.inet
+import scala.scalanative.posix.netdb._
+import scala.scalanative.posix.netdbOps._
+import scala.scalanative.posix.sys.ioctl._
+import scala.scalanative.posix.fcntl._
+import scala.scalanative.posix.sys.select._
+import scala.scalanative.posix.sys.selectOps._
+import scala.scalanative.posix.unistd.{close => cClose}
+import java.io.{FileDescriptor, IOException, OutputStream, InputStream}
+
+private[net] class PlainSocketImpl extends SocketImpl {
+
+  protected[net] var fd                   = new FileDescriptor
+  protected[net] var localport            = 0
+  protected[net] var address: InetAddress = null
+  protected[net] var port                 = 0
+
+  private var timeout = 0
+
+  override def getInetAddress: InetAddress       = address
+  override def getFileDescriptor: FileDescriptor = fd
+
+  override def create(streaming: Boolean): Unit = {
```

Comment:
It's a deprecated mode which makes `Socket` UDP instead of TCP. This can be delegated to `java.net.DatagramSocket` when it gets implemented.

https://github.com/asoltysik/scala-native/blob/1f24ccbefece8cbb4a21fe99ea6af6dbf9ee9ac1/javalib/src/main/scala/java/net/Socket.scala#L21-L25